packages/drivers/odsp-driver/src/test/socketTests/socketTests.spec.ts

```diff
@@ -0,0 +1,347 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { stub } from "sinon";
+import { v4 as uuid } from "uuid";
+import { IOdspResolvedUrl } from "@fluidframework/odsp-driver-definitions";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { ITelemetryLoggerExt, MockLogger, isFluidError } from "@fluidframework/telemetry-utils";
+import { DriverErrorTypes, IAnyDriverError } from "@fluidframework/driver-definitions";
+import { createOdspNetworkError } from "@fluidframework/odsp-doclib-utils";
+import { Socket } from "socket.io-client";
+import { EpochTracker } from "../../epochTracker";
+import { LocalPersistentCache } from "../../odspCache";
+import { getHashedDocumentId } from "../../odspPublicUtils";
+import { OdspDocumentDeltaConnection } from "../../odspDocumentDeltaConnection";
+import * as socketModule from "../../socketModule";
+import { ClientSocketMock } from "./socketMock";
+
+describe("OdspDocumentDeltaConnection tests", () => {
+	let tenantId = "tenantId";
+	let documentId = "documentId";
+	const token = "token";
+	const client: IClient = {
+		mode: "write",
+		scopes: ["doc:read", "doc:write"],
+		details: { capabilities: { interactive: true } },
+		permission: [],
+		user: { id: "userId" },
+	};
+	const webSocketUrl = "https://webSocketUrl";
+	let logger: ITelemetryLoggerExt;
+	const socketReferenceKeyPrefix = "prefix";
+	const siteUrl = "https://microsoft.sharepoint-df.com/siteUrl";
+	const driveId = "driveId";
+	const itemId = "itemId";
+	let epochTracker: EpochTracker;
+	let localCache: LocalPersistentCache;
+	let hashedDocumentId: string;
+	const resolvedUrl = {
+		siteUrl,
+		driveId,
+		itemId,
+		odspResolvedUrl: true,
+	} as any as IOdspResolvedUrl;
+	let socket: ClientSocketMock | undefined;
+
+	before(async () => {
+		hashedDocumentId = await getHashedDocumentId(driveId, itemId);
+	});
+
+	beforeEach(async () => {
+		logger = new MockLogger().toTelemetryLogger();
+		documentId = uuid();
+		tenantId = uuid();
+		// use null logger here as we expect errors
+		epochTracker = new EpochTracker(
+			localCache,
+			{
+				docId: hashedDocumentId,
+				resolvedUrl,
+			},
+			logger,
+		);
+	});
+
+	afterEach(async () => {
+		socket?.close();
+		await epochTracker.removeEntries().catch(() => {});
+	});
+
+	async function mockSocket<T>(_response: Socket, callback: () => Promise<T>): Promise<T> {
+		const geSocketCreationStub = stub(socketModule, "SocketIOClientStatic");
```

Comment:
did you mean get ? If not, what does ge stand for ?

Reply:
yes get:)